### PR TITLE
RPC 500 error handling

### PR DESCRIPTION
**Pull request by tingfeng-key**
RPC 500 error handling #150

### CODE REVIEW

File: lib/grammers-client/src/client/updates.rs [hunk 1]
```diff
@@ -124,6 +124,18 @@ impl Client {
                         message_box.end_channel_difference(&request, PrematureEndReason::Banned);
                         continue;
                     }
+                    Err(InvocationError::Rpc(rpc_error)) if rpc_error.code.eq(&500) => {
```
Comment (Lonami) [inline lib/grammers-client/src/client/updates.rs:127]: Could be just `== 500`.

**Review by Lonami — CHANGES_REQUESTED** [review]

**Comment by tingfeng-key** [comment]
I want to add a log to him, but I don't know what is more appropriate to output

**Comment by Lonami** [comment]
You can probably log a warning, since this error should hopefully be rare enough.

**Comment by tingfeng-key** [comment]
I have added it, you can see if you need to modify it

File: lib/grammers-client/src/client/updates.rs [hunk 2]
```diff
@@ -125,6 +125,7 @@ impl Client {
                         continue;
                     }
                     Err(InvocationError::Rpc(rpc_error)) if rpc_error.code == 500 => {
+                        log::warn!("rpc 500 err: {:#?}", rpc_error);
```
Comment (Lonami) [inline lib/grammers-client/src/client/updates.rs:128]:
```suggestion
                        log::warn!("Telegram is having internal issues: {:#?}", rpc_error);
```

**Pull request merged**